pallets/pallet-bonded-coins/src/traits.rs

```diff
@@ -80,21 +82,37 @@ pub trait ResetTeam<AccountId>: Inspect<AccountId> {
 	) -> DispatchResult;
 }
 
+/// Implementation of the back-ported ResetTeam trait for the assets pallet,
+/// relying on its `transfer_ownership` and `set_team` calls. Later versions of
+/// the assets pallet implement the original trait, so this is a stop-gap
+/// solution until we upgraded to at least Polkadot SDK version 1.13.0.
 impl<T, I: 'static> ResetTeam<AccountIdOf<T>> for AssetsPallet<T, I>
 where
 	T: AssetConfig<I>,
 	<<T as frame_system::Config>::Lookup as StaticLookup>::Source: From<AccountIdOf<T>>,
 {
 	fn reset_team(
 		id: Self::AssetId,
-		_owner: AccountIdOf<T>,
+		owner: AccountIdOf<T>,
 		admin: AccountIdOf<T>,
 		issuer: AccountIdOf<T>,
 		freezer: AccountIdOf<T>,
 	) -> DispatchResult {
-		let owner = AssetsPallet::<T, I>::owner(id.clone()).ok_or(DispatchError::Unavailable)?;
-		let origin = RawOrigin::Signed(owner);
-		AssetsPallet::<T, I>::set_team(origin.into(), id.into(), issuer.into(), admin.into(), freezer.into())
+		let current_owner = AssetsPallet::<T, I>::owner(id.clone()).ok_or(DispatchError::Unavailable)?;
```

Comment:
Do we need this check? Would not this be checked from within the `transfer_ownership` function? If yes, this is redundant. If not, why do we need it? Just so that we can run the `if`?

Reply:
We need the current owner to construct a SignedOrigin which is authorised to perform ownership changes. We can omit the check if the current owner is the new owner, but this would increase weights somewhat. We'd be reading the owner again, and performing the same check in the transfer_ownership function. But, yeah, not strictly required - should I get rid of the check?

Reply:
Repeated storage reads are cached anyway, so they don't incur in any additional weight. If the check is not strictly forbidden, I would remove it, also because if for any reason the underlying logic changes, this duplicated logic risks to go out of sync. So, yeah, if not strictly required I would remove it. If weight is a concern, it is not since reads are and only committed once.

Reply:
Ok, done in dc4f8f2cf7a1d86c39f7f075e7ed54a15b4d30c9